drivers/net/ethernet/intel/idpf/idpf.h

```diff
@@ -717,125 +275,96 @@ static inline void __iomem *idpf_get_reg_addr(struct idpf_adapter *adapter,
- *
- * Returns true or false if a particular feature is enabled.
- */
-static inline bool idpf_is_feature_ena(const struct idpf_vport *vport,
-				       netdev_features_t feature)
-{
-	return vport->netdev->features & feature;
-}
-
-/**
- * idpf_get_max_tx_hdr_size -- get the size of tx header
- * @adapter: Driver specific private structure
- */
-static inline u16 idpf_get_max_tx_hdr_size(struct idpf_adapter *adapter)
-{
-	return le16_to_cpu(adapter->caps.max_tx_hdr_size);
+	return (test_bit(IDPF_HR_RESET_IN_PROG, idpf_adapter_flags(adapter)) ||
+		test_bit(IDPF_HR_FUNC_RESET, idpf_adapter_flags(adapter)) ||
+		test_bit(IDPF_HR_DRV_LOAD, idpf_adapter_flags(adapter)));
 }
 
 /**
- * idpf_vport_ctrl_lock - Acquire the vport control lock
- * @netdev: Network interface device structure
+ * idpf_is_reset_detected - check if we were reset at some point
+ * @adapter: driver specific private structure
  *
- * This lock should be used by non-datapath code to protect against vport
- * destruction.
- */
-static inline void idpf_vport_ctrl_lock(struct net_device *netdev)
-{
-	struct idpf_netdev_priv *np = netdev_priv(netdev);
-
-	mutex_lock(&np->adapter->vport_ctrl_lock);
-}
-
-/**
- * idpf_vport_ctrl_unlock - Release the vport control lock
- * @netdev: Network interface device structure
+ * Returns true if we are either in reset currently or were previously reset.
  */
-static inline void idpf_vport_ctrl_unlock(struct net_device *netdev)
+static inline bool idpf_is_reset_detected(struct idpf_adapter *adapter)
 {
-	struct idpf_netdev_priv *np = netdev_priv(netdev);
+	if (!adapter->hw.arq)
+		return true;
 
-	mutex_unlock(&np->adapter->vport_ctrl_lock);
+	return !(readl(idpf_get_reg_addr(adapter, adapter->hw.arq->reg.len)) &
+		 adapter->hw.arq->reg.len_mask);
 }
 
-void idpf_statistics_task(struct work_struct *work);
-void idpf_init_task(struct work_struct *work);
 void idpf_service_task(struct work_struct *work);
 void idpf_mbx_task(struct work_struct *work);
 void idpf_vc_event_task(struct work_struct *work);
 void idpf_dev_ops_init(struct idpf_adapter *adapter);
 void idpf_vf_dev_ops_init(struct idpf_adapter *adapter);
 int idpf_intr_req(struct idpf_adapter *adapter);
 void idpf_intr_rel(struct idpf_adapter *adapter);
-u16 idpf_get_max_tx_hdr_size(struct idpf_adapter *adapter);
-int idpf_initiate_soft_reset(struct idpf_vport *vport,
-			     enum idpf_vport_reset_cause reset_cause);
-void idpf_deinit_task(struct idpf_adapter *adapter);
-int idpf_req_rel_vector_indexes(struct idpf_adapter *adapter,
-				u16 *q_vector_idxs,
-				struct idpf_vector_info *vec_info);
-void idpf_set_ethtool_ops(struct net_device *netdev);
-void idpf_vport_intr_write_itr(struct idpf_q_vector *q_vector,
-			       u16 itr, bool tx);
+
 int idpf_sriov_configure(struct pci_dev *pdev, int num_vfs);
 
 u8 idpf_vport_get_hsplit(const struct idpf_vport *vport);
 bool idpf_vport_set_hsplit(const struct idpf_vport *vport, u8 val);
+int idpf_init_dflt_mbx(struct idpf_adapter *adapter);
```

Comment:
vport hsplit functions above should be part of aux driver??

Reply:
Moved to idpf_vport.h